### ISSUE TITLE
else_body returns None?

### ISSUE THREAD

**Issue by hiker** [stfc/PSyclone#477]
Hi,

in 370eac165 a reviewer's comment was addressed and the return value of 'else_body' was changed from '[]' to 'None' if there was no else.

I know I am a bit late to the party ... but returning 'None' makes some coding more complicated than necessary. In my scripts I had:
```
for child in node.else_body:
```
and if there was no else body, the loop was just skipped.

Now with else_body returning None, this line becomes:
```
if node.else_body:
    for child in node.else_body:
```

Not too much of a deal tbh, but I wonder if we have a good reason for returning 'None' instead of []?  And if not ... would you object me changing this back in my next pull request?

**Comment by arporter** [comment]
I think I requested that change for consistency with the `if_body` method. In both cases these methods return a single `Schedule` node or `None` so there's no need to loop over `node.else_body`. Does that answer your question @hiker?

**Comment by hiker** [comment]
I don't think you are right that if_body can return None ... not according to the docs, nor can I create it by parsing code ... I think if_body _always_ returns a (potentially empty) schedule:
```
if (something) then      ! empty schedule as if_body
else
   bla
endif
```
while else_body might return an empty schedule:
```
if (something) then 
   bla
else     ! empty schedule as else_body
endif
```
or None:
```
if (something) then
  bla
endif      ! None as else_body
```
So in order to be consistent with if_body, it should return an empty schedule (or perhaps if_body should return None??) - though of course you can say that an if statement always has an if (not not necessarily an else) ... *sigh*

I admit it's academic (we need to distinguish between "no else" and "no else body" in order to return the equivalent code ... though do we need an else without body) - but my grief is that I now have to write:
```
        for child in node.if_body:
            if not is_blockable(child):
                return False
        if node.else_body:              # not required for if_body above
            for child in node.else_body:
                if not is_blockable(child):
                    return False
```
And I think this asymetry is annoying. If else_body would return [], we can still distinguish between "no else" and "else without a body", and can still write loops over the body without this if test.

**Comment by arporter** [comment]
Yes, you're right - the if_body should never return `None`, my mistake. However, the point I wanted to make is that neither of the methods return a list - they both return a `Schedule` object so I *think* your code should be:

    if not is_blockable(node.if_body):
        return False
    if node.else_body and not is_blockable(node.else_body):
        return False

this then captures the fact that, as you say, we're not guaranteed to have an `else` clause. What do you think?
